src/main/java/seedu/address/logic/commands/StatisticsCommand.java

```diff
@@ -18,38 +18,78 @@ public class StatisticsCommand extends Command {
 
     public static final String COMMAND_WORD = "statistics";
 
-
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        // create a filtered list of all the predicate persons
-        // then create iteration
         ObservableList<Person> lastShownList = model.getFilteredPersonList();
         FilteredList<Person> fullContactList = new FilteredList<>(lastShownList);
-        int numOfApplicants = 0;
-        HashMap<JobCode, Integer> jobDict = new HashMap<JobCode, Integer>();
-        HashMap<Tag, Integer> tagDict = new HashMap<Tag, Integer>();
+
+        HashMap<JobCode, JobCodeStatistics> jobStatisticsMap = new HashMap<>();
+        calculateJobCodeStatistics(fullContactList, jobStatisticsMap);
+        String statisticsMessage = buildStatisticsMessage(jobStatisticsMap);
+        return new CommandResult(statisticsMessage);
+    }
+
+    private void calculateJobCodeStatistics(FilteredList<Person> fullContactList,
```

Comment:
The loop structure here is clear and effective for accumulating statistics. To enhance readability, consider adding comments to explain the logic behind retrieving and updating job statistics. Also, ensure that person.getTag() will not return null to avoid potential NullPointerExceptions.